vizro-core/src/vizro/models/_components/form/dropdown.py

```diff
@@ -88,10 +88,12 @@ def validate_multi(cls, multi, values):
             raise ValueError("Please set multi=True if providing a list of default values.")
         return multi
 
+    # AM comment: please remove build_static and change into __call__ in all places unless you think there's
+    # a good reason not to do so.
     def __call__(self, options):
-        return self._build_static(options)
-
-    def _build_static(self, options):
+        # AM comment: this is the main confusing thing about the current approach I think: every time we run
+        # __call__ we override the value set below, which sounds wrong because we say that we never change the value.
+        # From what I remember this is a workaround for the Dash bug? Maybe add a comment explaining this.
```

Comment:
It's idea to set `self.value` only once (and then to reuse it) because dash bug enforce us to return the same value every time so that the persistence from the storage can be applied to the UI component. If the `self.value` is set (and it's always set for the dynamic component), we don't overwrite it anymore, as we can see from:
`value = self.value if self.value is not None else default_value`.

And as you already said (I like the comment), once we have a better control of the pre_build ordering, we will set the `self.value` in the pre_build phase and reuse it for static and dynamic component, without a need for `default_value` variable.